### DISCUSSION TITLE
No successful execution of the hello_world.py and waypoints.py. referring to #610

### DISCUSSION THREAD

**Discussion by aarsht7** [USC-ACTLab/crazyswarm#631] [Q&A]
Referring to the end of #610 ,

[GideonLombardo](https://github.com/GideonLombardo) says,

>  I have been able to run the hello_world.py script successfully. However, I do have some questions. The script does not work every time. For example, when the crazyflie coordinate axes do not align with the world axes, the script will sometimes fail. The crazyflie axes defined in RVIZ will sometimes shake and flicker. I have a feeling the inconsistency in stable flight is arising from the "shaky" axes. Is there a known reason why these things are happening? Please let me know.

I am facing the similar issues as GideonLombardo. I am using `vrpn` with unique markers and it works some times it doesn't most of the time. When it doesn't work, it will just flip over. I tried with `send_position_only = True`. One strange thing happens when I run the `hover_swarm.launch` is, it is logging warning `No updated pose for motion capture object 'cf#'` same as [kaushikbalasundar](https://github.com/kaushikbalasundar) has in #564 . But it doesn't seem to be an issue as [kaushikbalasundar](https://github.com/kaushikbalasundar) mentioned in the #564  the pose do shown up in rviz. It doesn't flicker much or at least it is not noticeable. I do not know what the problem can be. I have one more thing to ask, if we are using `send_position_only = True` then the flickering of the axis (angular deviation) should not affect the actual behavior of the flight, right?

---
Edit-1
I am not sure but it can be the issue of balancing. I found [this](https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyflie-2-x/#unbalanced) and [this](https://www.bitcraze.io/documentation/tutorials/balancing-propellers/) on their official website. I will try it out before my next testing session will update the flipping issue of the CF here.

---
Edit-2
There is no issue of the propeller or weight imbalance. I tried it flying the crazyflie with loco system and it is quite stable. but it is not working well with the optitrack motion capture system. I see loss of data while running the `mocap_helper.launch`. it skips the every second message all the time. I am not sure if that can be an issue.

---

**Comment by whoenig** [comment]
If you don't get reliable data from your motion capture system that is a problem. Usually that happens if you use an older version of Motive. We tested with Motive 2.3 and 3.0 (beta).

**Reply by aarsht7** [reply]
Unfortunately, I have motive 2.2. And I tried use [this](https://github.com/mje-nz/natnet_ros) in hope to get reliable data but it is only tested with motive 2.0, 2.1 and Ros kinetic, melodic. I am using ros noetic + motive 2.2 and I am facing some issues with that. I can change the ros distro but not the motive version. It feels I am running out of luck haha.

**Reply by whoenig** [reply]
You can try using "optitrack_closed_source" rather than "optitrack" (requires a custom re-build, since it doesn't build by default, see https://github.com/USC-ACTLab/crazyswarm/discussions/522).

**Reply by aarsht7** [reply]
I tried `optitrack_closed_source` now it does not give the error but it seem is not getting the data of the marker either.
I have changed the broadcasting in the Motive application and my configuration is as follows,

My configuration in the Motive app is as follows.

    Broadcast Frame Data : On
    Local Interface : 192.168.0.100
    Labeled Markers : On
    Unlabeled Markers : On
    Asset Markers : Off
    Rigid Bodies : Off
    Up Axis : Z Up
    Remote Trigger : Off
    Transmission Type : Multicast


When I run `roslaunch crazyswarm mocap_helper.launch` I do not see any points (marker data).
```
process[mocap_helper-2]: started with pid [370415]
frame 0:
    points:
    rigidBodies:
frame 1:
    points:
    rigidBodies:
frame 2:
    points:
    rigidBodies:
frame 3:
    points:
    rigidBodies:
frame 4:
    points:
    rigidBodies:
```

I thought it might be the issue of the printing output. But when I try to use kalman estimator, it seem it is not considering any data of the motion capture as my /cf#/pose is just keep getting increased when I echo it. And when it move the CF fast enough by hands to get it reset (I was testing what initial position it sets), it reset to the (0, 0, 0) instead of the given initial position in the crazyflies.yaml 

I have also tried to print the Number of the markers [from here](https://github.com/IMRCLab/libmotioncapture/blob/27cd890bbd3c99ca28e7115edaf72dfe0299be20/src/optitrack_closed_source.cpp#L47) but it is zero always.

I am not sure if is it suppose to do that?

---
Edit - 1

I tried to use the [NatNet SDK](https://optitrack.com/support/downloads/developer-tools.html) from OptiTrack and it seem working fine. I can see the exact location of the markers.

**Reply by whoenig** [reply]
optitrack_closed_souce is exactly the NatNet SDK. I assume in motive, you do not have any assets enabled?

**Reply by aarsht7** [reply]
Yes, I concluded the same that `optitrack_closed_source` is the NatNet SDK. But still I do not know if something is going wrong. The thing is it can detect the individual rigid bodies and the markers from the body when I enable the rigid body publishing. But When I try with single marker or labeled/unlabeled markers it fails. I will try modifying few things in the `optitrack_closed_source.cpp` and will try to have a look if it works.

> I assume in motive, you do not have any assets enabled?

I am not sure what do you mean by assets. If you mean rigid body, then no I do not have any rigid body enabled.
